### PR TITLE
added version 3.1 changelog

### PR DESCRIPTION
**Pull request by jasonwee**

### CODE REVIEW

File: spec/src/main/asciidoc/messaging-spec.adoc [hunk 1]
```diff
@@ -6825,6 +6825,17 @@ TextMessage which throw JMSException.
...
+* https://github.com/jakartaee/messaging/pull/311[issue_311] Bump `maven-bundle-plugin` version from 4.2.1 to 5.1.4
+* https://github.com/jakartaee/messaging/pull/306[issue_306] Remove useless parentheses
+* https://github.com/jakartaee/messaging/pull/298[issue_298] Repeatable annotation for the connection factory
+* https://github.com/jakartaee/messaging/pull/296[issue_296] Update `JMSDestinationDefinition`
+* https://github.com/jakartaee/messaging/pull/292[issue_292] Introduce new Ctors to set linked exception on creation
+* https://github.com/jakartaee/messaging/pull/291[issue_291] Version bump from 3.0 to 3.1
+* https://github.com/jakartaee/messaging/pull/289[issue_289] Typo fixed
+* https://github.com/jakartaee/messaging/pull/287[issue_287] Comment fixed
+* https://github.com/jakartaee/messaging/pull/284[issue_284] Minor updates to spec copyright and version referenced in javadoc
+
```
Comment (pzygielo) [inline spec/src/main/asciidoc/messaging-spec.adoc:6838]: What about
- #308 or #309

File: spec/src/main/asciidoc/messaging-spec.adoc [hunk 2]
```diff
@@ -6825,6 +6825,17 @@ TextMessage which throw JMSException.
 [appendix]
 == Change history
 
+=== Version 3.1
+* https://github.com/jakartaee/messaging/pull/311[issue_311] Bump `maven-bundle-plugin` version from 4.2.1 to 5.1.4
+* https://github.com/jakartaee/messaging/pull/306[issue_306] Remove useless parentheses
+* https://github.com/jakartaee/messaging/pull/298[issue_298] Repeatable annotation for the connection factory
```
Comment (OndroMih) [inline spec/src/main/asciidoc/messaging-spec.adoc:6831]: I think ti's worth to mention both annotations separately
```suggestion
* https://github.com/jakartaee/messaging/pull/298[issue_298] Repeatable annotation for the connection factory
* https://github.com/jakartaee/messaging/pull/298[issue_298] Repeatable annotation for the destination definition
```

File: spec/src/main/asciidoc/messaging-spec.adoc [hunk 3]
```diff
@@ -6825,6 +6825,17 @@ TextMessage which throw JMSException.
...
+* https://github.com/jakartaee/messaging/pull/296[issue_296] Update `JMSDestinationDefinition`
```
Comment (OndroMih) [inline spec/src/main/asciidoc/messaging-spec.adoc:6832]:
```suggestion
* https://github.com/jakartaee/messaging/pull/296[issue_296] Update `JMSDestinationDefinition`Javadoc comment
```

File: spec/src/main/asciidoc/messaging-spec.adoc (hunk 1 above)
Comment (OndroMih) [reply spec/src/main/asciidoc/messaging-spec.adoc:6838]: That one wasn't marked with the 3.1 milestone, I've fixed that.
Also this one: https://github.com/jakartaee/messaging/pull/313

File: spec/src/main/asciidoc/messaging-spec.adoc [hunk 4]
```diff
@@ -6825,6 +6825,17 @@ TextMessage which throw JMSException.
...
+* https://github.com/jakartaee/messaging/pull/292[issue_292] Introduce new Ctors to set linked exception on creation
+* https://github.com/jakartaee/messaging/pull/291[issue_291] Version bump from 3.0 to 3.1
```
Comment (OndroMih) [inline spec/src/main/asciidoc/messaging-spec.adoc:6834]: This one can be removed, it's only a change in the pom.xml maven file.

File: spec/src/main/asciidoc/messaging-spec.adoc [hunk 5]
```diff
@@ -6825,6 +6825,17 @@ TextMessage which throw JMSException.
 [appendix]
 == Change history
 
+=== Version 3.1
+* https://github.com/jakartaee/messaging/pull/311[issue_311] Bump `maven-bundle-plugin` version from 4.2.1 to 5.1.4
+* https://github.com/jakartaee/messaging/pull/306[issue_306] Remove useless parentheses
+* https://github.com/jakartaee/messaging/pull/298[issue_298] Repeatable annotation for the connection factory
+* https://github.com/jakartaee/messaging/pull/296[issue_296] Update `JMSDestinationDefinition`
+* https://github.com/jakartaee/messaging/pull/292[issue_292] Introduce new Ctors to set linked exception on creation
```
Comment (OndroMih) [inline spec/src/main/asciidoc/messaging-spec.adoc:6833]:
```suggestion
* https://github.com/jakartaee/messaging/pull/292[issue_292] New constructors in exception classes
```

**Comment by OndroMih** [comment]
Hi @jasonwee, thanks a lot for this!

I left my suggestions.
Please also add those 2 PRs mentioned above.

And I would create 2 lists, first for API-related changes (only repeatable annotations and new constructors in exceptions) and the other changes. For example:

=== Version 3.1

==== API changes

* ...
* ...

==== Other improvements

* ...
* ...

**Review by OndroMih — CHANGES_REQUESTED** [review]

**Review by OndroMih — APPROVED** [review]

**Pull request merged**